lib/autoInject.js

```diff
@@ -106,22 +108,25 @@ export default function autoInject(tasks, callback) {
 
     forOwn(tasks, function (taskFn, key) {
         var params;
+        var fnIsAsync = isAsync(taskFn);
 
         if (isArray(taskFn)) {
             params = taskFn.slice(0, -1);
             taskFn = taskFn[taskFn.length - 1];
 
             newTasks[key] = params.concat(params.length > 0 ? newTask : taskFn);
-        } else if (taskFn.length === 1) {
+        } else if ((!fnIsAsync && taskFn.length === 1) ||
+            (fnIsAsync && taskFn.length === 0)) {
```

Comment:
Actually, I should probably factor this condition out into a `hasNoDeps` variable.